package.py

```diff
@@ -1,8 +1,10 @@
 name = "hiero"
 title = "Hiero"
 version = "0.2.3-dev.1"
+app_host_name = "hiero"
```

Comment:
Where is that used, out of curiosity ?
I've seen most of the other addons do not define this yet.

Reply:
I'm curious about this to. I can only find [this in the backend](https://github.com/ynput/ayon-backend/blob/044b74403ea883524f3ff5c5985cb1d501278480/ayon_server/addons/addon.py#L61) and other than that [ayon-wrap](https://github.com/ynput/ayon-wrap/blob/c717a424f01d73820d01dd146749ad184da2c92d/package.py#L4) is the only other repo on `ynput` organization mentioning this in the `package.py` **however** that interestingly enough does not specify `ayon_server_version`.

Looking at the backend - it feels like this shouldn't be in `package.py` but in `server` on the `BaseServerAddon` since there I at least see it specified and used?

However, also the [`get_app_host_names`](https://github.com/search?q=org%3Aynput%20get_app_host_names&type=code) function appears unused? Seems like a mystery _why_ is needed - and also that'd make the actual _testing_ of this theoretical as well since there's no usage (yet?).

Would be good to include the details on the issue why it's needed? And I'm also curious to hear why it needs to be in `package.py` instead of on the server addon?

Reply:
Wow you pretty much summed up my whole investigation which ended up with similar wondering :)

Reply:
It is what we consider as "host name". But server backend, before `1.1.2`, did validate if that value changed between addon versions, and if it did, it caused crash of addon discovery. So we couldn't add it to addons at the time, because it would cause huge production issues. Now, it is not validated and it is safe to add it.

It should be in `package.py`, predefined attributes from `package.py` are auto-applied to `BaseServerAddon` automatically (like `name`, `version`, `title`, `services`, `app_host_name`, etc.).

Reply:
> It should be in package.py. Predefined attributes from package.py are auto-applied to BaseServerAddon automatically (like name, version, title, services, app_host_name, etc.).

Auto-applied when and where? :)

Reply:
https://github.com/ynput/ayon-backend/blob/develop/ayon_server/addons/definition.py#L113